docs/decisions/0031-feature-branch-strategy.md

```diff
@@ -0,0 +1,93 @@
+- **Timely Feature Integration**: Small, incremental pull requests allow for quicker reviews and faster integration of features into the feature branch and make it easier to merge down into main as the code was already previously reviewed. This timeliness ensures that features are merged and ready for deployment sooner, improving the responsiveness to changes.
+
+## Considered Options
+
+- Community Feature Branch Strategy
+
+### Community Feature Branch Strategy
+
+As soon we identify that contributors are willing to take/create a Feature Issue as a potential connector implementation, we will create a new branch for that feature.
+
+The contributor(s) will then be one of the responsibles to incrementally add the majority of changes through small Pull Requests to the feature branch under our supervision and review process.
+
+This strategy involves creating a separate branch in the repository for each new big feature, like connectors. This isolation means that changes are made in a controlled environment without affecting the main branch.
+
+We may also engage in the development and changes to the feature branch when needed, the changes and full or co-authorship on the PRs will be tracked and properly referred into the Release Notes.
+
+#### Pros and Cons
+
+- Good, because it allows for focused development on one feature at a time.
+- Good, because it promotes smaller, incremental Pull Requests (PRs), simplifying review processes.
+- Good, because it reduces the risk of major bugs being merged into the main branch.
+- Good, because it makes the process of integrating features into the main branch easier and faster.
+- Bad, potentially, if not managed properly, as it can lead to outdated branches if not regularly synchronized with the main branch.
+
+## Validation
+
+The effectiveness of the feature branch strategy will be validated through regular code reviews, successful CI/CD pipeline runs, and feedback from the development team.
+
+## Connector/Model Priorities
+
+Currently we are looking for community support on the following models
+
+The support on the below can be either achieved creating a practical example using one of the existing Connectors against one of this models or providing a new Connector that supports a platform that hosts one of the models below:
+
+| Model Name | Local Support | Platforms                            | Connectors                                    |
+| ---------- | ------------- | ------------------------------------ | --------------------------------------------- |
+| Gpt-4      | No            | OpenAI, Azure Open AI                | OpenAI                                        |
+| Phi-2      | Yes           | Azure OpenAI, Hugging Face, LMStudio | OpenAI, HuggingFace, LMStudio\*\*             |
```

Comment:
Do we need a connector for both `OpenAI` and `LMStudio`?

This article https://www.developerscantina.com/p/semantic-kernel-open-source-llms/ shows how to use our current OpenAI connector to call a model hosted locally using LMStudio.

If a developer builds a solution using a model deployed locally on LMStudio what will their path to deploying this to production be?

Reply:
We might at some point, I'm leaving this in the documentation because our OpenAI connector is strongly reliant into Azure SDK and any change to it would eventually break LMStudio support right away. 

Having a dedicated connector to LMStudio allow it to be specific and support customization that may not be possible in closed/cloud LLMs.

Reply:
Added observation regarding LMStudio connector may not be desired at first.